clinica/pipelines/dwi_preprocessing_using_t1/dwi_preprocessing_using_t1_utils.py

```diff
@@ -212,87 +216,185 @@ def print_end_pipeline(image_id, final_file):
     print_end_image(image_id)
 
 
-def prepare_reference_b0(in_dwi, in_bval, in_bvec, low_bval=5, working_directory=None):
+def prepare_reference_b0(
+    in_dwi: str,
+    in_bval: str,
+    in_bvec: str,
+    low_bval: int = 5,
+    working_directory=None,
+):
```

Comment:
This function is the entry point of the whole B0 computation. It is the function that is used, as an interface, in the Nipype workflow. This means a few things:

- type hints cannot be something else than builtins because the function has to be self-contained (for example, I can't put `Optional[str]` for the argument `working_directory` because nipype will complain that Optional is not defined.
- It is for this reason that I kept the previous signature with tuples of strings as input and output while I changed the downstream code to use more complex abstractions.
- We can change the variable names to match the ones I've proposed in this PR, we just need to update the interface call in the DWI workflow.

One possibility is to wrap the implementation within a task function which is only responsible for parsing the inputs and outputs. Something that would look like this probably:

```python
def prepare_reference_b0_task(
    dwi_filename: str,
    b_values_filename: str,
    b_vectors_filename: str,
    b_value_threshold: float = 5.0,
    working_directory = None,
):
    """Task called be Nipype."""
    from pathlib import Path
    from clinica.pipelines... import prepare_reference_b0

    if working_directory:
        working_directory = Path(working_directory)

    b0_reference_filename, merged_dwi_dataset = prepare_reference_b0(
        DWIDataset(
            dwi=dwi_filename: str,
            b_values=b_values_filename,
            b_vectors=b_vectors_filename,
        ),
        b_value_threshold,
        working_directory,
    )

    return str(b0_reference_filename), *(str(_) for _ in merged_dwi_dataset)

def prepare_reference_b0(
    dwi_dataset: DWIDataset,
    b_value_threshold: float = 5.0,
    working_directory: Optional[Path]=None,
) -> Tuple[Path, DWIDataset]:
    """Internal implementation.
    Returns the path to the reference B0 image and the DWIDataset obtained after
    merging the reference B0 into the original DWI dataset.
    """
    ....
```

Does this make sense ?